### ISSUE TITLE
Crash on start (Aborted (core dumped))

### ISSUE THREAD

**Issue by neutrinus** [mltframework/shotcut#1475]
When running ShotCut there is a loading screen and then it dies. Everytime. Both from snap and appimage:

```
marek@blacky:~/Pobrane$ ./shotcut-linux-x86_64-230929.AppImage 
libDeckLinkAPI.so: cannot open shared object file: No such file or directory
Aborted (core dumped)
marek@blacky:~/Pobrane$ 
```

```
marek@blacky:~$ whereis shotcut
shotcut: /snap/bin/shotcut
marek@blacky:~$ shotcut
libDeckLinkAPI.so: cannot open shared object file: No such file or directory
Aborted (core dumped)
marek@blacky:~$ 
```

Then I tried to investigate more:
```marek@blacky:~$ cd /snap/shotcut/current/Shotcut.app/
marek@blacky:/snap/shotcut/current/Shotcut.app$ . source-me 
marek@blacky:/snap/shotcut/current/Shotcut.app$ gdb bin/shotcut
gdb: /snap/shotcut/current/Shotcut.app/lib/libtinfo.so.6: version `NCURSES6_TINFO_6.2.20211010' not found (required by /lib/x86_64-linux-gnu/libncursesw.so.6)
marek@blacky:/snap/shotcut/current/Shotcut.app$ 
```

```marek@blacky:~/Pobrane$ gdb shotcut-linux-x86_64-230929.AppImage
GNU gdb (Ubuntu 14.0.50.20230907-0ubuntu1) 14.0.50.20230907-git
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from shotcut-linux-x86_64-230929.AppImage...
This GDB supports auto-downloading debuginfo from the following URLs:
  <https://debuginfod.ubuntu.com>
Enable debuginfod for this session? (y or [n]) 
Debuginfod has been disabled.
To make this setting permanent, add 'set debuginfod enabled off' to .gdbinit.
(No debugging symbols found in shotcut-linux-x86_64-230929.AppImage)
(gdb) run
Starting program: /home/marek/Pobrane/shotcut-linux-x86_64-230929.AppImage 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[Detaching after fork from child process 87925]
process 87922 is executing new program: /tmp/.mount_shotcuvGKY7v/AppRun
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
process 87922 is executing new program: /usr/bin/dash
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[Detaching after fork from child process 87930]
[Detaching after fork from child process 87931]
[Detaching after vfork from child process 87932]
libDeckLinkAPI.so: cannot open shared object file: No such file or directory
Aborted (core dumped)
[Inferior 1 (process 87922) exited with code 0206]
(gdb) bt
No stack.
(gdb) 
```
(yup, I don't know what I'm doing)

My system is amd64  Ubuntu 23.10 64bit, Gnome+Wayland on lenovo x1-carbon gen6

**The last working appimage for me is v23.06.14.**

**Comment by ddennedy** [comment]
Related forum threads:
https://forum.shotcut.org/t/all-appimages-are-broken-refuse-to-load-any-video-files-on-all-linux-versions-with-fresh-kernel-updates/40994
and
https://forum.shotcut.org/t/shotcut-fails-to-load-any-video-file/40938

Does removing the file `/snap/shotcut/current/Shotcut.app/lib/libtinfo.so.6` help?

**Comment by neutrinus** [comment]
It looks like you can not remove files from snap: (squashfs is read-only system)
```
marek@blacky:~$ LANG=C sudo rm /snap/shotcut/current/Shotcut.app/lib/libtinfo.so.6 
rm: cannot remove '/snap/shotcut/current/Shotcut.app/lib/libtinfo.so.6': Read-only file system
marek@blacky:~$ 
```

Trying the portable way:
```
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ ./shotcut 
libDeckLinkAPI.so: cannot open shared object file: No such file or directory
Aborted (core dumped)
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ 
```


As pointed in the related forum threads, maybe there is something wrong with my config files?  (there is nothing valuable there for me, so deleting is much easier way)
```
rm -rf ~/.config/Meltytech/ ~/.local/share/Meltytech/
``` 
and that didn't fix it (snap, Appimage, portable still broken). But that give me the idea where to look for logs!

So lets run a portable again:
```
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ rm -rf ~/.config/Meltytech/ ~/.local/share/Meltytech/
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ ./shotcut 
libDeckLinkAPI.so: cannot open shared object file: No such file or directory
Aborted (core dumped)
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ tail ~/.local/share/Meltytech/Shotcut/shotcut-log.txt 
[Info   ] <OpenGLVideoWidget::initialize> OpenGL threaded? true 
[Info   ] <OpenGLVideoWidget::initialize> OpenGL ES? false 
[Info   ] <OpenGLVideoWidget::initialize> OpenGL maximum texture size = 16384 
[Info   ] <OpenGLVideoWidget::initialize> OpenGL maximum viewport size = 16384 x 16384 
[Debug  ] <OpenGLVideoWidget::initialize> end 
[Debug  ] <Mlt::VideoWidget::initialize> begin 
[Debug  ] <Mlt::VideoWidget::initialize> end 
[Debug  ] <NewProjectFolder::showEvent> m_profile "" action.data "" 
[Info   ] <Util::isMemoryLow> available RAM =  9382508 KB 
[Fatal  ] <> Cannot mix incompatible Qt library (6.4.2) with this library (6.4.3)
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ 
marek@blacky:~$ dpkg -l | grep libqt6core6
ii  libqt6core6:amd64                              6.4.2+dfsg-18                           amd64        Qt 6 core module
marek@blacky:~$ 

```

Just to be ultra-clear, the Fatal thing happens in all 3 scenarios: appimage, snap and portable.


Can I do anything more to help investigate it further?

**Comment by ddennedy** [comment]
Shotcut is using Qt 6.4.3 (upgraded in July), but it bundles it in Shotcut.app/lib and the startup script makes it prefer that lib directory over the system's. I think there could be something missing in the bundle like libQt6QmlCore.so.6. I will look into more.

**Comment by ddennedy** [comment]
I have an Ubuntu 20.04 system with Qt 6.4.2 from Qt installer, but its `lib` dir is added to my `LD_LIBRARY_PATH` environment variable. The launch script does not overwrite `LD_LIBRARY_PATH` but instead prepends to it. This launches and opens video just fine in AppImage, portable, and Snap.

**Comment by ddennedy** [comment]
I tested in Ubuntu 23.10, and I did not get the fatal Qt 6 library conflict after `sudo apt install qml-qt6`, but I did get a backwards compatibility problem with libcairo:
https://forum.shotcut.org/t/shotcut-fails-to-load-any-video-file/40938/13

This library is no longer bundled because it was causing problems upon migrating to Qt 6: 1ee9cade5e

**Comment by ddennedy** [comment]
I removed from the portable `Shotcut.app/lib/libpixman-1.so.0` and now it works. I need to exclude this lib from the bundle as well.

**Comment by neutrinus** [comment]
Yup, on freshly installed Ubuntu 23.10 the portable works like it should. Thank you!

But on my ubuntu it still crashes.

So, let's try with the portable `shotcut-linux-x86_64-230929.txz `

```
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ QT_DEBUG_PLUGINS=1 ./shotcut
[...]
[Debug  ] <> "/home/marek/Pobrane/Shotcut/Shotcut.app/lib/qml/QtQuick/Dialogs/libqtquickdialogsplugin.so" loaded library
[Debug  ] <> checking directory path "/home/marek/Pobrane/Shotcut/Shotcut.app/lib/qt6/wayland-decoration-client" ...
[Debug  ] <> looking at "/home/marek/Pobrane/Shotcut/Shotcut.app/lib/qt6/wayland-decoration-client/libbradient.so"
[Debug  ] <> Found metadata in lib /home/marek/Pobrane/Shotcut/Shotcut.app/lib/qt6/wayland-decoration-client/libbradient.so, metadata=
{
    "IID": "org.qt-project.Qt.WaylandClient.QWaylandDecorationFactoryInterface.5.4",
    "MetaData": {
        "Keys": [
            "bradient"
        ]
    },
    "archlevel": 1,
    "className": "QWaylandBradientDecorationPlugin",
    "debug": false,
    "version": 394240
}


[Debug  ] <> Got keys from plugin meta data QList("bradient")
[Debug  ] <> checking directory path "/home/marek/Pobrane/Shotcut/Shotcut.app/bin/wayland-decoration-client" ...
[Debug  ] <> "/home/marek/Pobrane/Shotcut/Shotcut.app/lib/qt6/wayland-decoration-client/libbradient.so" loaded library
Aborted (core dumped)
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ 
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ rm /home/marek/Pobrane/Shotcut/Shotcut.app/lib/qt6/wayland-decoration-client/libbradient.so

```
And after this rm, it no longer crashes! 

(there is no window decoration thou)

**Comment by neutrinus** [comment]
I know I am here like a crazy racoon, but maybe it will help somehow. My idea now is that in this libbradient is something linked that causes this `incompatible Qt library` error.

```
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ . source-me 
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ ldd /home/marek/Pobrane/Shotcut/Shotcut.app/lib/qt6/wayland-decoration-client/libbradient.so
/home/marek/Pobrane/Shotcut/Shotcut.app/lib/qt6/wayland-decoration-client/libbradient.so: /lib/x86_64-linux-gnu/libQt6WaylandClient.so.6: no version information available (required by /home/marek/Pobrane/Shotcut/Shotcut.app/lib/qt6/wayland-decoration-client/libbradient.so)
	linux-vdso.so.1 (0x00007ffef299d000)
	libQt6WaylandClient.so.6 => /lib/x86_64-linux-gnu/libQt6WaylandClient.so.6 (0x00007fa09e049000)
	libQt6Gui.so.6 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libQt6Gui.so.6 (0x00007fa09d600000)
	libQt6Core.so.6 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libQt6Core.so.6 (0x00007fa09ce00000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fa09ca00000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa09c600000)
	libwayland-client.so.0 => /lib/x86_64-linux-gnu/libwayland-client.so.0 (0x00007fa09e036000)
	libwayland-cursor.so.0 => /lib/x86_64-linux-gnu/libwayland-cursor.so.0 (0x00007fa09e02c000)
	libxkbcommon.so.0 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libxkbcommon.so.0 (0x00007fa09dfea000)
	libEGL.so.1 => /lib/x86_64-linux-gnu/libEGL.so.1 (0x00007fa09dfd7000)
	libfontconfig.so.1 => /lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007fa09df89000)
	libX11.so.6 => /lib/x86_64-linux-gnu/libX11.so.6 (0x00007fa09d4c2000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fa09df82000)
	libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007fa09ccb9000)
	libQt6DBus.so.6 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libQt6DBus.so.6 (0x00007fa09c931000)
	libGLX.so.0 => /lib/x86_64-linux-gnu/libGLX.so.0 (0x00007fa09df50000)
	libOpenGL.so.0 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libOpenGL.so.0 (0x00007fa09df24000)
	libpng16.so.16 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libpng16.so.16 (0x00007fa09deec000)
	libharfbuzz.so.0 => /lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007fa09c82a000)
	libfreetype.so.6 => /lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007fa09c534000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fa09d4a3000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa09dee5000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa09c449000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fa09d47f000)
	libicui18n.so.66 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libicui18n.so.66 (0x00007fa09c000000)
	libicuuc.so.66 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libicuuc.so.66 (0x00007fa09be1a000)
	libdouble-conversion.so.3 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libdouble-conversion.so.3 (0x00007fa09d469000)
	libpcre2-16.so.0 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libpcre2-16.so.0 (0x00007fa09c3c5000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fa09e14f000)
	libffi.so.8 => /lib/x86_64-linux-gnu/libffi.so.8 (0x00007fa09ded8000)
	libGLdispatch.so.0 => /lib/x86_64-linux-gnu/libGLdispatch.so.0 (0x00007fa09c30d000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007fa09d43e000)
	libxcb.so.1 => /lib/x86_64-linux-gnu/libxcb.so.1 (0x00007fa09d414000)
	libpcre2-8.so.0 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libpcre2-8.so.0 (0x00007fa09bd89000)
	libdbus-1.so.3 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libdbus-1.so.3 (0x00007fa09bd38000)
	libgraphite2.so.3 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libgraphite2.so.3 (0x00007fa09cc8c000)
	libbz2.so.1.0 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libbz2.so.1.0 (0x00007fa09cc79000)
	libbrotlidec.so.1 => /lib/x86_64-linux-gnu/libbrotlidec.so.1 (0x00007fa09cc6c000)
	libicudata.so.66 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libicudata.so.66 (0x00007fa09a200000)
	libXau.so.6 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libXau.so.6 (0x00007fa09d40c000)
	libXdmcp.so.6 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libXdmcp.so.6 (0x00007fa09c822000)
	libsystemd.so.0 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libsystemd.so.0 (0x00007fa09a151000)
	libbrotlicommon.so.1 => /lib/x86_64-linux-gnu/libbrotlicommon.so.1 (0x00007fa09bd15000)
	libbsd.so.0 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libbsd.so.0 (0x00007fa09bcfb000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fa09c81d000)
	liblzma.so.5 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/liblzma.so.5 (0x00007fa09bcd2000)
	liblz4.so.1 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/liblz4.so.1 (0x00007fa09a130000)
	libgcrypt.so.20 => /home/marek/Pobrane/Shotcut/Shotcut.app/lib/libgcrypt.so.20 (0x00007fa09a012000)
	libgpg-error.so.0 => /lib/x86_64-linux-gnu/libgpg-error.so.0 (0x00007fa099fec000)
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$
marek@blacky:~/Pobrane/Shotcut/Shotcut.app$ ldd /home/marek/Pobrane/Shotcut/Shotcut.app/lib/qt6/wayland-decoration-client/libbradient.so | grep -v marek
	linux-vdso.so.1 (0x00007ffeeabe9000)
	libQt6WaylandClient.so.6 => /lib/x86_64-linux-gnu/libQt6WaylandClient.so.6 (0x00007fed542bb000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fed52c00000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fed52800000)
	libwayland-client.so.0 => /lib/x86_64-linux-gnu/libwayland-client.so.0 (0x00007fed542a8000)
	libwayland-cursor.so.0 => /lib/x86_64-linux-gnu/libwayland-cursor.so.0 (0x00007fed5429e000)
	libEGL.so.1 => /lib/x86_64-linux-gnu/libEGL.so.1 (0x00007fed54249000)
	libfontconfig.so.1 => /lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007fed541fb000)
	libX11.so.6 => /lib/x86_64-linux-gnu/libX11.so.6 (0x00007fed536c2000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fed541f4000)
	libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007fed52eb9000)
	libGLX.so.0 => /lib/x86_64-linux-gnu/libGLX.so.0 (0x00007fed540f3000)
	libharfbuzz.so.0 => /lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007fed52af9000)
	libfreetype.so.6 => /lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007fed52a2d000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fed5363f000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fed540ec000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fed52715000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fed5361b000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fed543c1000)
	libffi.so.8 => /lib/x86_64-linux-gnu/libffi.so.8 (0x00007fed540df000)
	libGLdispatch.so.0 => /lib/x86_64-linux-gnu/libGLdispatch.so.0 (0x00007fed520de000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007fed52e78000)
	libxcb.so.1 => /lib/x86_64-linux-gnu/libxcb.so.1 (0x00007fed520b4000)
	libbrotlidec.so.1 => /lib/x86_64-linux-gnu/libbrotlidec.so.1 (0x00007fed5360e000)
	libbrotlicommon.so.1 => /lib/x86_64-linux-gnu/libbrotlicommon.so.1 (0x00007fed51ed3000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fed52e6d000)
	libgpg-error.so.0 => /lib/x86_64-linux-gnu/libgpg-error.so.0 (0x00007fed50258000)

```

So, there is only one QT library linked from system and not from portable: `libQt6WaylandClient.so.6`. Is this the cause of this issue?

**Comment by ddennedy** [comment]
Yes, I think you found the problem, and I can fix it in our builds if so. To verify, please unzip the Qt 6.4.3 version of  `libQt6WaylandClient.so.6` to `Shotcut.app/lib` and restore `libbradient.so` (maybe extract the portable archive again). Thanks for your help.

**Comment by neutrinus** [comment]
It works with the supplied file. No crash on startup. 

Thank you, it was a pleasure!

**Comment by ddennedy** [comment]
Including `libQt6WaylandClient.so.6` is causing failures to start on Ubuntu 22.04, which is more important as LTS. 

https://forum.shotcut.org/t/segmentation-fault-on-startup-on-linux-v23-11-29/41895/11?

I just installed Ubuntu 23.10 in a virtual machine, which uses wayland by default, and I needed to also remove this library as well to get it to work. Therefore, I am removing this library for the next version 23.12
